# 1**. Cricket - Test Matches Analysis**
Cricket is one of the most popular spectator sports in the world with viewership in billions spread across the globe. Test cricket format is the toughest of all in cricket format due to longer time spend, unlimited bow and others. By Test Mathces analysis, the quality of a Player can be easily understandable. This is because if you want to survive in the longest format you will be needing a lot more qualities than you need in the shorter formats. For this format, cricketers need to be patient, aggressive, good readers of the game and a good thinker. Only a few people in the world are this qualified and talented.
In this notebook, I will do a comprehensive analysis of the Test Cricket Matches by comparing over different categories. I'll look for insights in the data to devise strategies to findout world most qualified and talented playes in Test Cricket.

In [2]:
# importing data analysis libraries
import pandas as pd
import numpy as np


**How to Import an Excel File into Python using Pandas**


1.   need to capture the full path where the Excel file is stored on your computer.
2.   if there is a specific Excel sheet to import, need to specify it under sheet_name.

**Error handling**


> **ImportError: Missing optional dependency ‘xlrd’**
-need to install openpyxl for .xlsx files (using PIP install)



In [3]:
# read data set from excel sheet 
df = pd.read_excel('/content/drive/MyDrive/test_cricket.xlsx', sheet_name = 'wickets')

In [4]:
# display the first few rows of the data
display(df.head(10))


Player       Span  Mat  Inns  ...  Econ    SR   5  10
0  M Muralitharan (ICC/SL)  1992-2010  133   230  ...  2.47  55.0  67  22
1           SK Warne (AUS)  1992-2007  145   273  ...  2.65  57.4  37  10
2         A Kumble (INDIA)  1990-2008  132   236  ...  2.69  65.9  35   8
3        JM Anderson (ENG)  2003-2021  162   301  ...  2.83  56.3  30   3
4         GD McGrath (AUS)  1993-2007  124   243  ...  2.49  51.9  29   3
5          SCJ Broad (ENG)  2007-2021  148   272  ...  2.92  56.8  18   3
6            CA Walsh (WI)  1984-2001  132   242  ...  2.53  57.8  22   3
7            DW Steyn (SA)  2004-2019   93   171  ...  3.24  42.3  26   5
8      N Kapil Dev (INDIA)  1978-1994  131   227  ...  2.78  63.9  23   2
9        HMRKB Herath (SL)  1999-2018   93   170  ...  2.80  60.0  34   9

[10 rows x 14 columns]

In [5]:
# display the all features or variables or all columns name 
df.columns


Index(['Player',   'Span',    'Mat',   'Inns',  'Balls',   'Runs',   'Wkts',
          'BBI',    'BBM',    'Ave',   'Econ',     'SR',        5,       10],
      dtype='object')

# **2.Given Feature variables**


1. Player = Name of a Player name
2. Span =  total time between starting the test_cricket match and ending the test criket match of a Player
3. Mat =  Total number of Match played by a player
4. Inns =  Total number of Innings played by a player (One Match = 2 innings_)
5. Balls = Total number of Ball throw by a player
6. Runs = Total runs have been given by a bowler
7. Wkts = Total wicket has been taken by a bowler in his carrier
8. BBI = BBI stands for Best Bowling in Innings  for a player
9. BBM = BBM stands for Best Bowling in Match and gives the combined score over 2 or more innings in one match.
10. Ave = The total number of runs divided by the total number of innings in which the batsman was out.
11. Econ = a player's economy rate is the average number of runs they have conceded per over bowled.Economy refers to the runs you concede in each over of your spell. It is the ratio of total runs conceded to the total overs bowled 
12. SR = Batting strike rate (s/r) is defined for a batsman as the average number of runs scored per 100 balls faced. The higher the strike rate, the more effective a batsman is at scoring quickly.
13. 5 = The total number of 5 wickets has taken in a match by a bowler
14. 10 =  The total number of 10 wickets has taken in a Match by a bowler




In [6]:
# shape of the dataset (total 79 rows and 14 columns in the dataset)
df.shape

(79, 14)

In [7]:
#data statistics, data types and missing value check
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79 entries, 0 to 78
Data columns (total 14 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  79 non-null     object 
 1   Span    79 non-null     object 
 2   Mat     79 non-null     int64  
 3   Inns    79 non-null     int64  
 4   Balls   79 non-null     int64  
 5   Runs    79 non-null     int64  
 6   Wkts    79 non-null     int64  
 7   BBI     79 non-null     object 
 8   BBM     79 non-null     object 
 9   Ave     79 non-null     float64
 10  Econ    79 non-null     float64
 11  SR      79 non-null     float64
 12  5       79 non-null     int64  
 13  10      79 non-null     int64  
dtypes: float64(3), int64(7), object(4)
memory usage: 8.8+ KB


**3. Observations:**

* Number of observations (rows) = 79
* Number of columns = 14
* The dataset has no missing values
* memory usage 8.8 kilobyte

In [8]:
# check the data types ( string (object, integer, float)of each variables)
df.dtypes

Player     object
Span       object
Mat         int64
Inns        int64
Balls       int64
Runs        int64
Wkts        int64
BBI        object
BBM        object
Ave       float64
Econ      float64
SR        float64
5           int64
10          int64
dtype: object

In [9]:
df.isna().any() # missing value check, no missing value in data

Player    False
Span      False
Mat       False
Inns      False
Balls     False
Runs      False
Wkts      False
BBI       False
BBM       False
Ave       False
Econ      False
SR        False
5         False
10        False
dtype: bool

In [10]:
# checking data statistics
df.describe()

Mat        Inns         Balls  ...         SR          5         10
count   79.000000   79.000000     79.000000  ...  79.000000  79.000000  79.000000
mean    80.101266  144.797468  18630.303797  ...  59.187342  16.354430   2.797468
std     28.537692   51.042310   7190.036515  ...   9.349337   9.642372   3.235935
min     37.000000   67.000000   8785.000000  ...  41.200000   3.000000   0.000000
25%     60.500000  110.000000  13580.000000  ...  53.300000   9.500000   1.000000
50%     71.000000  129.000000  16498.000000  ...  57.400000  14.000000   2.000000
75%     93.000000  169.000000  21742.500000  ...  63.950000  20.500000   3.500000
max    166.000000  301.000000  44039.000000  ...  91.900000  67.000000  22.000000

[8 rows x 10 columns]

# **data statistics :**

* Minumum Match played by a bowler is 37
* The average wicket taken by a bowler in the dataset is 317

* The Highest Strike rate is 91 for a bowler

* The average runs has been given by a bowler is 8595 

* The most expensive player is  3.46 (from the Economy rate)

* the highest number of innigs played by a player is 301


In [11]:
# rename a column in the dataset
df= df.rename(columns={'Mat':"Match", 
                       "Inns":"Innigns",
                       'BBI':'Best_Bowling_in_Innings',
                       'BBM':'Bowling_in_Match',
                       'SR':'Strike_rate',
                       5 :'5_wkts', 
                       10 :'10_wkts'})
df

Player       Span  Match  ...  Strike_rate  5_wkts  10_wkts
0   M Muralitharan (ICC/SL)  1992-2010    133  ...         55.0      67       22
1            SK Warne (AUS)  1992-2007    145  ...         57.4      37       10
2          A Kumble (INDIA)  1990-2008    132  ...         65.9      35        8
3         JM Anderson (ENG)  2003-2021    162  ...         56.3      30        3
4          GD McGrath (AUS)  1993-2007    124  ...         51.9      29        3
..                      ...        ...    ...  ...          ...     ...      ...
74        SCG MacGill (AUS)  1998-2008     44  ...         54.0      12        2
75    Saqlain Mushtaq (PAK)  1995-2004     49  ...         67.6      13        3
76         AME Roberts (WI)  1974-1983     47  ...         55.1      11        2
77            JA Snow (ENG)  1965-1976     49  ...         59.5       8        1
78         JR Thomson (AUS)  1972-1985     51  ...         52.6       8        0

[79 rows x 14 columns]

In [14]:
# remove a column
df1 = df.loc[:,"Player":"5_wkts"] # remove the 10_wkts column from the dataset
df1

Player       Span  Match  ...  Econ  Strike_rate  5_wkts
0   M Muralitharan (ICC/SL)  1992-2010    133  ...  2.47         55.0      67
1            SK Warne (AUS)  1992-2007    145  ...  2.65         57.4      37
2          A Kumble (INDIA)  1990-2008    132  ...  2.69         65.9      35
3         JM Anderson (ENG)  2003-2021    162  ...  2.83         56.3      30
4          GD McGrath (AUS)  1993-2007    124  ...  2.49         51.9      29
..                      ...        ...    ...  ...   ...          ...     ...
74        SCG MacGill (AUS)  1998-2008     44  ...  3.22         54.0      12
75    Saqlain Mushtaq (PAK)  1995-2004     49  ...  2.64         67.6      13
76         AME Roberts (WI)  1974-1983     47  ...  2.78         55.1      11
77            JA Snow (ENG)  1965-1976     49  ...  2.68         59.5       8
78         JR Thomson (AUS)  1972-1985     51  ...  3.18         52.6       8

[79 rows x 13 columns]

In [16]:
# another way to remove a column from the dataset

col_names = ['Player',   'Span',    'Mat',   'Inns',  'Balls',   'Runs']
df_usecols = pd.read_excel('/content/drive/MyDrive/test_cricket.xlsx', sheet_name = 'wickets', usecols=col_names)

display(df_usecols.head(10))

Player       Span  Mat  Inns  Balls   Runs
0  M Muralitharan (ICC/SL)  1992-2010  133   230  44039  18180
1           SK Warne (AUS)  1992-2007  145   273  40705  17995
2         A Kumble (INDIA)  1990-2008  132   236  40850  18355
3        JM Anderson (ENG)  2003-2021  162   301  34791  16457
4         GD McGrath (AUS)  1993-2007  124   243  29248  12186
5          SCJ Broad (ENG)  2007-2021  148   272  29713  14502
6            CA Walsh (WI)  1984-2001  132   242  30019  12688
7            DW Steyn (SA)  2004-2019   93   171  18608  10077
8      N Kapil Dev (INDIA)  1978-1994  131   227  27740  12867
9        HMRKB Herath (SL)  1999-2018   93   170  25993  12157

In [17]:
# rading the first n number of rows
df = pd.read_excel('/content/drive/MyDrive/test_cricket.xlsx', sheet_name = 'wickets', nrows=50)

display(df.head(15))
print(df.shape)

Player       Span  Mat  Inns  ...  Econ    SR   5  10
0   M Muralitharan (ICC/SL)  1992-2010  133   230  ...  2.47  55.0  67  22
1            SK Warne (AUS)  1992-2007  145   273  ...  2.65  57.4  37  10
2          A Kumble (INDIA)  1990-2008  132   236  ...  2.69  65.9  35   8
3         JM Anderson (ENG)  2003-2021  162   301  ...  2.83  56.3  30   3
4          GD McGrath (AUS)  1993-2007  124   243  ...  2.49  51.9  29   3
5           SCJ Broad (ENG)  2007-2021  148   272  ...  2.92  56.8  18   3
6             CA Walsh (WI)  1984-2001  132   242  ...  2.53  57.8  22   3
7             DW Steyn (SA)  2004-2019   93   171  ...  3.24  42.3  26   5
8       N Kapil Dev (INDIA)  1978-1994  131   227  ...  2.78  63.9  23   2
9         HMRKB Herath (SL)  1999-2018   93   170  ...  2.80  60.0  34   9
10       Sir RJ Hadlee (NZ)  1973-1990   86   150  ...  2.63  50.8  36   9
11          SM Pollock (SA)  1995-2008  108   202  ...  2.39  57.8  16   1
12  Harbhajan Singh (INDIA)  1998-2015  103   190  ...  2.84  68.5  25   5
13        Wasim Akram (PAK)  1985-2002  104   181  ...  2.59  54.6  25   5
14         R Ashwin (INDIA)  2011-2021   79   148  ...  2.80  52.4  30   7

[15 rows x 14 columns]

(50, 14)
